src/helpers/sql.ts

```diff
@@ -0,0 +1,30 @@
+import { SelectQueryBuilder } from "kysely";
+import { parseRange } from "./content-range";
+import { Database } from "../types";
+
+export async function executeQuery<T extends keyof Database>(
+  query: SelectQueryBuilder<Database, T, {}>,
+  rangeRequest?: string,
+) {
+  const parsedRange = parseRange(rangeRequest);
+
+  const data = await query
+    .selectAll()
+    .offset(parsedRange.from)
+    .limit(parsedRange.limit)
+    .execute();
+
+  let range;
+  if (parsedRange.entity) {
+    const [{ count }] = await query
+      .select((eb) => eb.fn.countAll().as("count"))
+      .execute();
+
+    range = `${parsedRange.entity}=${parsedRange.from}-${parsedRange.to}/${count}`;
+  }
```

Comment:
ah yes here we go!  Same comment as on the other PR :smile: 

```suggestion
  
  if (!parsedRange.entity) return { data }
 
  const [{ count }] = await query
      .select((eb) => eb.fn.countAll().as("count"))
      .execute();

  const range = `${parsedRange.entity}=${parsedRange.from}-${parsedRange.to}/${count}`;
 ```